src/sinks/util/http.rs

```diff
@@ -17,20 +21,92 @@ use tower_hyper::client::Client;
 use tracing::field;
 use tracing_tower::{InstrumentableService, InstrumentedService};
 
-pub type RequestBuilder = Box<dyn Fn(Vec<u8>) -> hyper::Request<Vec<u8>> + Sync + Send>;
 pub type Response = hyper::Response<Bytes>;
 pub type Error = hyper::Error;
 
+pub trait HttpSink: Send + Sync + 'static {
+    type Input;
+    type Output;
+
+    fn encode_event(&self, event: Event) -> Option<Self::Input>;
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>>;
+}
+
+pub struct BatchedHttpSink<T, B: Batch>
+where
+    B::Output: Clone,
+{
+    sink: Arc<T>,
+    inner: BatchSink<B, TowerBatchedSink<B::Output, HttpRetryLogic, B, HttpService<B::Output>>>,
```

Comment:
Are we wrapping two sinks?

Reply:
I understand now that one is actually not a sink, but I will leave the evidence of my initial confusion.